diff --git a/es/useXAgent/index.d.ts b/es/useXAgent/index.d.ts
index 1060cc2eec41d3e15ab8892a73bdaea9a2b0a80e..85bc84d44cf2dace89f020f0066d1bc477942fdc 100644
--- a/es/useXAgent/index.d.ts
+++ b/es/useXAgent/index.d.ts
@@ -17,7 +17,7 @@ export interface XAgentConfigPreset {
 export interface XAgentConfigCustom<Message> {
     request?: RequestFn<Message>;
 }
-export type XAgentConfig<Message> = Partial<XAgentConfigPreset> & XAgentConfigCustom<Message>;
+export type XAgentConfig<Message> = Partial<XAgentConfigPreset> & XAgentConfigCustom<Message> & {customParams: array<any>};
 /** This is a wrap class to avoid developer can get too much on origin object */
 export declare class XAgent<Message = string> {
     config: XAgentConfig<Message>;
diff --git a/es/useXAgent/index.js b/es/useXAgent/index.js
index 0e827decc3c1707548ec3a90a8f9bf5dc6f9c6cf..562f54ffe3508cac8b7225082f92c057774e4e67 100644
--- a/es/useXAgent/index.js
+++ b/es/useXAgent/index.js
@@ -53,6 +53,7 @@ export class XAgent {
 export default function useXAgent(config) {
   const {
     request,
+    customParams = [],
     ...restConfig
   } = config;
   return React.useMemo(() => [new XAgent({
@@ -62,5 +63,5 @@ export default function useXAgent(config) {
       dangerouslyApiKey: restConfig.dangerouslyApiKey
     }).create,
     ...restConfig
-  })], []);
+  })], [...customParams]);
 }
